garmin_fit.cc

```diff
@@ -395,198 +412,256 @@ GarminFitFormat::fit_parse_data(const fit_message_def& def, int time_offset)
       debug_print(7,"%s: parsing field %d\n", MYNAME, i);
     }
     const fit_field_t& f = def.fields.at(i);
-    uint32_t val = fit_read_field(f);
-    if (f.id == kFieldTimestamp) {
-      if (global_opts.debug_level >= 7) {
-        debug_print(7,"%s: parsing fit data: timestamp=%d\n", MYNAME, val);
-      }
-      timestamp = val;
-      // if the timestamp is < 0x10000000, this value represents
-      // system time; to convert it to UTC, add the global utc offset to it
-      if (timestamp < 0x10000000) {
-        timestamp += fit_data.global_utc_offset;
-      }
-      fit_data.last_timestamp = timestamp;
-    } else {
+    if (f.type == 7) {
+      QString val = fit_getstring(f.size);
       switch (def.global_id) {
-      case kIdDeviceSettings: // device settings message
+      case kIdLocations:
         switch (f.id) {
-        case kFieldGlobalUtcOffset:
+        case kFieldLocationName:
           if (global_opts.debug_level >= 7) {
```

Comment:
Perhaps it is better to teach fit_read_field to handle f.type 7 data and return a QVariant.  Then the handling of kIdLocations can be handled in the def.global_id switch block kIdLocations case below along with latitude, longitude, altitude.